packages/shared/src/lib/core/wallet/actions/send/createEvmChainToEvmChainTransaction.ts

```diff
@@ -76,6 +79,7 @@ export function createEvmChainToEvmChainTransaction(
 
 function getDataForTransaction(
```

Comment:
It would be cleaner to split this function into two. This way you don't have so many nested layers.